crates/language-server/src/backend/db.rs

```diff
@@ -0,0 +1,58 @@
+use common::InputDb;
+
+use hir::{HirDb, LowerHirDb, SpannedHirDb};
+use hir_analysis::HirAnalysisDb;
+use salsa::{ParallelDatabase, Snapshot};
+
+#[salsa::jar(db = LanguageServerDb)]
+pub struct Jar(crate::functionality::diagnostics::file_line_starts);
+
+pub trait LanguageServerDb:
+    salsa::DbWithJar<Jar> + HirAnalysisDb + HirDb + LowerHirDb + SpannedHirDb + InputDb
+{
+}
+
+impl<DB> LanguageServerDb for DB where
+    DB: Sized + salsa::DbWithJar<Jar> + HirAnalysisDb + HirDb + LowerHirDb + SpannedHirDb + InputDb
+{
+}
+
+#[salsa::db(
+    common::Jar,
+    hir::Jar,
+    hir::LowerJar,
+    hir::SpannedJar,
+    hir_analysis::Jar,
+    Jar
+)]
+pub struct LanguageServerDatabase {
+    storage: salsa::Storage<Self>,
+}
+
+impl LanguageServerDatabase {
+    pub fn as_language_server_db(&self) -> &dyn LanguageServerDb {
+        <Self as salsa::DbWithJar<Jar>>::as_jar_db::<'_>(self)
+    }
+}
+
+impl salsa::Database for LanguageServerDatabase {
+    fn salsa_event(&self, _: salsa::Event) {}
```

Comment:
The logger has been showing some salsa events already even without implementing this, I guess from the other crates?  I actually disabled them temporarily [here](https://github.com/micahscopes/fe/blob/04b3fc2fa0c4019eb6d99cd5464597821c0d5e27/crates/language-server/src/logger.rs#L55-L59) in order to focus on messages from the language-server crate.

It could be cool to have separate outputs for these or some other way of toggling.